packages/datadog-instrumentations/src/helpers/instrument.js

```diff
@@ -119,3 +120,49 @@ if (semver.satisfies(process.versions.node, '>=16.0.0')) {
     return exports.bindAsyncResource.call(ar, fn, thisArg)
   }
 }
+
+exports.bindEventEmitter = function bindEventEmitter (emitter) {
```

Comment:
Do we actually need this? When I wrote the current implementation, I just blindly bound all emitters but it was likely a mistake and not needed. Is there an actual use case solved by this?

Reply:
I believe its only needed for the 'on' emitter for Command. So I think we can only bind that and leave the rest alone

Reply:
`on` isn't an emitter, it's a function.

You definitely need all the event emitter methods, because you don't know which ones will be used. You need to bind any listener added, to specific events, and it's quite a bit easier just to bind them for any event. I doubt there's any use in filtering them. We also need to not break removing listeners so this is pretty much needed as-is.

We need it in this case because MySql gives us an EventEmitter. @khanayan123 I know you just removed this, but please restore it.

Reply:
The original code for this was from my misunderstanding of context propagation from years ago and should be removed. Event listeners are not like callbacks and promises and shouldn't be treated as such. Callbacks and promises happen after an operation have finished, so it makes sense that the handler is executed in the context of the parent that was active before the operation was started. Event listeners on the other hand are executed at an arbitrary point in time and can be executed multiple times. As such, they should run in the context of the emitter, not the listener. This doesn't mean that there can't be cases where this is incorrect because the library doesn't run its code with the correct AsyncResource, but this should be handled only in the necessary edge cases and eventually corrected in the library itself.